### PR TITLE
Cell counting plugin

### PR DESCRIPTION
**Pull request by mkurc-ant**
This is a plugin that allows to return to the TCL interpreter number of selected modules, cells or wires. It is used in synthesis script for QuickLogic FPGAs.

### CODE REVIEW

File: get_count-plugin/get_count.cc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#include "kernel/rtlil.h"
+
+#ifndef YS_OVERRIDE
+#define YS_OVERRIDE override
+#endif
+
+USING_YOSYS_NAMESPACE
+PRIVATE_NAMESPACE_BEGIN
+
+void register_in_tcl_interpreter(const std::string& command) {
```
Comment (tmichalak) [inline get_count-plugin/get_count.cc:31]: While working on other plugins I came to realize that this is not needed as this is called during `yosys -import`. Remoe the function and its use.

File: get_count-plugin/get_count.cc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    enum class ObjectType {
+        NONE,
+        MODULE,
+        CELL,
+        WIRE
+    };
+
+    GetCount () :
+        Pass("get_count", "Returns count of various selected object types to the TCL interpreter") {
+            register_in_tcl_interpreter(pass_name);
```
Comment (tmichalak) [inline get_count-plugin/get_count.cc:48]: remove

File: get_count-plugin/get_count.cc [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            break;
+        default:
+            log_assert(false);
+        }
+
+        // Return the value as string to the TCL interpreter
+        std::string value = std::to_string(count);
+
+        Tcl_Obj* tclStr = Tcl_NewStringObj(value.c_str(), value.size());
+        Tcl_ListObjAppendElement(tclInterp, tclList, tclStr);
```
Comment (tmichalak) [inline get_count-plugin/get_count.cc:131]: I don't think you need a tcl list here:
```
Tcl_Obj* tclStr = Tcl_NewStringObj(value.c_str(), -1);
Tcl_SetObjResult(tclInterp, tclStr);
```
should be enough.

File: get_count-plugin/get_count.cc [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        else {
+            log_error("Object type not specified!\n");
+        }
+
+        extra_args(a_Args, 2, a_Design);
+
+        // Get the TCL interpreter
+        Tcl_Interp* tclInterp = yosys_get_tcl_interp();
+        Tcl_Obj*    tclList = Tcl_NewListObj(0, NULL);
```
Comment (tmichalak) [inline get_count-plugin/get_count.cc:98]: I don't think you need a tcl list, look comment below.

File: get_count-plugin/tests/simple/script.tcl [hunk 5]
```diff
@@ -0,0 +1,30 @@
+yosys plugin -i ../../get_count.so
```
Comment (tmichalak) [inline get_count-plugin/tests/simple/script.tcl:1]: the path shouldn't require any `../..`

**Review by tmichalak — CHANGES_REQUESTED** [review]
Looks good in general. Several comments in.

File: get_count-plugin/get_count.cc (hunk 1 above)
Comment (mkurc-ant) [reply get_count-plugin/get_count.cc]: Removed.

File: get_count-plugin/get_count.cc (hunk 2 above)
Comment (mkurc-ant) [reply get_count-plugin/get_count.cc]: Removed.

File: get_count-plugin/get_count.cc (hunk 3 above)
Comment (mkurc-ant) [reply get_count-plugin/get_count.cc]: Thanks, that works too. Fixed.

File: get_count-plugin/get_count.cc (hunk 4 above)
Comment (mkurc-ant) [reply get_count-plugin/get_count.cc]: Removed.

File: get_count-plugin/tests/simple/script.tcl (hunk 5 above)
Comment (mkurc-ant) [reply get_count-plugin/tests/simple/script.tcl]: Fixed.

Comment (tmichalak) [reply get_count-plugin/tests/simple/script.tcl:1]: Oh, now I see. `plugin -i name.so` looks in the specified directory and falls back to Yosys's shared folder.
I don't think we want `install` to depend on `test`. The tests are run in CI after `install` which copies all .so to Yosys's shared folder thus `plugin -i get_count.so` will work in CI.  
With your Makefile conda_packages will run the test before installing in contrast to other plugins.
I don't think this is needed.
My suggestions:
1. Don't call test before install
2. Change `plugin -i ../../get_count.so` to `plugin -i get_count.so`. For completeness as both options will work in CI. However, what if your case requires another plugin? It won't be available before install.

Comment (mkurc-ant) [reply get_count-plugin/tests/simple/script.tcl]: Ok. Now I understand. Fixed.

**Review by tmichalak — APPROVED** [review]
LGTM, will merge on green CI.

**Pull request merged**